app/models/section.rb

```diff
@@ -181,10 +188,14 @@ def all_editions
 
 private
 
-  attr_reader :slug_generator, :editions, :latest_edition
+  attr_reader :slug_generator, :latest_edition, :previous_edition
 
   def published_edition
-    most_recent_non_draft = editions.reject(&:draft?).last
+    most_recent_non_draft = if !latest_edition.draft?
+                              latest_edition
+                            elsif previous_edition && !previous_edition.draft?
+                              previous_edition
+                            end
 
     if most_recent_non_draft && most_recent_non_draft.published?
       most_recent_non_draft
```

Comment:
Observation: I haven't looked at it in detail, but I suspect there are some similarities between the logic in this method and [`Manual#current_published_version`](https://github.com/alphagov/manuals-publisher/blob/1947ca0d0cc264bde50bc542f95dd3a07de2cee5/app/models/manual.rb#L310-L328).